### PR TITLE
Expose FactRetrieverEngine from tech-insights-backend

### PR DESCRIPTION
**Pull request by Xantier**
## Hey, I just made a Pull Request!

Modify Tech insight initialization to expose FactRetrieverEngine. Enables users to trigger fact retrieval manually or reschedule retrievers on runtime.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [n/a] Tests for new functionality and regression tests for bug fixes
- [n/a] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/tech-insights-backend/api-report.md [hunk 1]
```diff
@@ -92,6 +113,7 @@ export type TechInsightsContext<
 > = {
   factChecker?: FactChecker<CheckType, CheckResultType>;
   persistenceContext: PersistenceContext;
+  factRetrieverEngine: FactRetrieverEngine;
```
Comment (freben) [inline plugins/tech-insights-backend/api-report.md:116]: You really would like to expose just a slim interface here, right, not the entire implementation class? Maybe even something with just a subset of the engine capabilities. The current code seems to invite the possibility of creating engines of your own and running scheduleI() etc which wasn't the intent, right?

Comment (Xantier) [reply plugins/tech-insights-backend/api-report.md:116]: Yeah, I think it makes sense to slim it down a little. We do want the possibility to reschedule the fact retrievers, tasks implementation being idempotent we can just add them as tasks again. Also the possibility to run a retriever manually with the already exposed method is a good to have. I'll wrap this into an interface and export that instead.

Comment (Xantier) [reply plugins/tech-insights-backend/api-report.md:103]: I've modified the PR to expose only an interface of the engine.

**Review by freben — APPROVED** [review]
Alright!

**Pull request merged**